addons/mail/static/src/models/chat_window/chat_window.js

```diff
@@ -340,16 +372,20 @@ function factory(dependencies) {
                 'threadDisplayName',
             ],
         }),
-        thread: many2one('mail.thread', {
-            related: 'threadViewer.thread',
-        }),
+        thread: many2one('mail.thread'),
         threadDisplayName: attr({
             related: 'thread.displayName',
         }),
         threadFoldState: attr({
             related: 'thread.foldState',
         }),
         threadViewer: one2one('mail.thread_viewer', {
+            compute: '_computeThreadViewer',
+            dependencies: [
+                'isVisible',
+                'thread',
+                'threadViewer',
+            ],
```

Comment:
Should consider to also remove the thread viewer when folded. It might simplify a lot of conditions especially in the component.

Reply:
It's tricky as scroll position is fucked up when you restore the chat window (a scroll event just at restore save a new position). Any idea on how to fix that, should we handle the remove on folded.